### PR TITLE
(c-r v0.2.0) *: update CRD manifest file names to <full group>_<resource>_crd.yaml 

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** CRD manifest name changes from #1642.

**Motivation for the change:** a CRD manifest can contain multiple API versions. Naming must change to reflect only the group and kind represented by that CRD.

### CODE REVIEW

**Comment by estroz** [comment]
Depends on #1696 being merged into refactor branch.

File: internal/pkg/scaffold/crd.go [hunk 1]
```diff
@@ -105,8 +101,12 @@ func (s *CRD) CustomRender() ([]byte, error) {
 
 		// controller-tools generates crd file names with no _crd.yaml suffix:
 		// <group>_<version>_<kind>.yaml.
-		path := strings.Replace(filepath.Base(i.Path), "_crd.yaml", ".yaml", 1)
-		b, err := afero.ReadFile(fs, path)
+		genPath := fmt.Sprintf("%s_%s_%s.yaml",
+			s.Resource.GoImportGroup,
+			s.Resource.Version,
+			s.Resource.LowerKind,
```
Comment (joelanford) [inline internal/pkg/scaffold/crd.go]: I could totally be missing something, but is this the correct path?

I'm looking at this: https://github.com/kubernetes-sigs/controller-tools/blob/46695abfb89bb6fcd972edf4f925915434726008/pkg/crd/gen.go#L77

Comment (estroz) [reply internal/pkg/scaffold/crd.go]: It is, but this PR doesn't bump c-t, just the CRD file name change.

File: doc/user/olm-catalog/generating-a-csv.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Operator SDK projects have an expected [project layout][doc-project-layout]. In
 * Roles: `role.yaml`
 * Deployments: `operator.yaml`
 * Custom Resources (CR's): `crds/<group>_<version>_<kind>_cr.yaml`
-* Custom Resource Definitions (CRD's): `crds/<group>_<version>_<kind>_crd.yaml`.
+* Custom Resource Definitions (CRD's): `crds/<group>_<resource>.yaml`.
```
Comment (hasbro17) [inline doc/user/olm-catalog/generating-a-csv.md]:
```suggestion
* Custom Resource Definitions (CRD's): `crds/<group>_<resource>_crd.yaml`.
```
We still have the `_crd` suffix.

File: CHANGELOG.md [hunk 3]
```diff
@@ -29,6 +29,7 @@
       // New
       err = r.client.List(context.TODO(), podList, client.UseListOptions(listOps))
       ```
+- CRD file names were previously of the form `<group>_<version>_<kind>_crd.yaml`. Now that CRD manifest `spec.version` is deprecated in favor of `spec.versions`, i.e. multiple versions can be specified in one CRD, CRD file names have the form `<full group>_<resource>_crd.yaml`. `<resource>` is the plural lower-case CRD Kind found at `spec.names.plural`.
```
Comment (hasbro17) [inline CHANGELOG.md]: Can we point out the difference between group vs full group here e.g: `foo.bar.com` is the full group and `foo` is the shortened group.
Remember only we make that distinction in the SDK.
The [k8s API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md) refer to it as just group.
```
Group names are typically in domain name form.
When choosing a group name, we recommend selecting a subdomain your group or organization owns, such as "widget.mycompany.com".
Group names must be lower case and be valid DNS subdomains
```

**Pull request merged**